### ISSUE TITLE
PowerSignOptimizer not available on TPUv2

### ISSUE THREAD

**Issue by Matt-Hicks-Bose** [tensorflow/tensorflow#19481]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: TPUv2-8
- **TensorFlow installed from (source or binary)**: VM disk image as configured by ctpu
- **TensorFlow version (use command below)**: v1.8.0-0-g93bc2e2072
- **Python version**: 2.7.13
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: Attempt to execute ```tf.contrib.opt.PowerSignOptimizer()``` on a TPUv2

### Describe the problem
The Powersign optimizer ```tf.contrib.opt.PowerSignOptimizer()``` does not have a OpKernel for the XLA_TPU_JIT device.

### Source code / logs
```
NotFoundError (see above for traceback): No registered 'ResourceApplyPowerSign' OpKernel for XLA_TPU_JIT devices compatible with node PowerSignOptimizer/update_dense/bias/ResourceApplyPowerSign = ResourceApplyPowerSign[T=DT_FLOAT, _class=["loc:@PowerSignOptimizer/update_dense/bias/Read/ReadVariableOp"], use_locking=false, _device="/device:TPU_REPLICATED_CORE"](_arg106, _arg10, PowerSignOptimizer/learning_rate, PowerSignOptimizer/logbase, PowerSignOptimizer/sign_decay, PowerSignOptimizer/beta, CrossReplicaSum_95)
        .  Registered:  device='CPU'; T in [DT_DOUBLE]
  device='CPU'; T in [DT_FLOAT]
  device='CPU'; T in [DT_BFLOAT16]
  device='CPU'; T in [DT_HALF]
```

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 76 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 93 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 108 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 123 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tatatodd** [comment]
Apologies for the long delay, the op was implemented a while ago (but after you filed this issue), and is here:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/tf2xla/kernels/training_ops.cc#L904
